### ISSUE TITLE
Is it possible to develop a function for uploading pictures, similar to the way pictures are placed in WordDocument?

### ISSUE THREAD

**Issue by rosyhickey** [vkbo/novelWriter#2115]
I hope to add a new function to save pictures in the left navigation of the editor, because I usually collect some information to facilitate my conception when writing novels. Some information is in the format of pictures, which I cannot save in my novel project, but can only save in my computer. If it can be saved in the novel project, it will be more convenient for my creation.
Thank you for checking！

**Comment by vkbo** [comment]
Hi, as stated in #815, novelWriter isn't meant to handle extensive research, Personally, I do my research using Obsidian. I only keep plot and character notes and background in novelWriter.

That said, there is a feature request for attachments, in #1060, and I do intend to allow adding external links to the project tree. Maybe even embedding them in the project folder. The latter makes more sense when I add the option to save the project as a single file to make it more portable,

> I hope to add a new function to save pictures in the left navigation of the editor

Do you mean you intend to develop it? If so, we need to discuss implementation. I'm currently rewriting the project tree using a model/view implementation instead, so there's no point developing on the current main branch.

**Comment by vkbo** [comment]
I'm leaving the ticket open for now, in case you want to respond, but since it is otherwise a duplicate of #1060, I will close one of them soon, depending on which one has the most details.

**Comment by rosyhickey** [comment]
Yes, it is somewhat similar to #1060, but my idea is whether it is possible to insert a picture directly in the middle of the text like a Word document, instead of an attachment?(Similar to the image below)
![screenshot_2024-11-22_16-59-35](https://github.com/user-attachments/assets/3799232a-a05c-4257-a40c-6753d8dfc165)
I am sorry that I am not a professional developer, I just put forward the applicable opinions as a user, thank you for your reply.

**Comment by vkbo** [comment]
> I saw the content of #815, and maybe it's a good idea as a plain text editor, except that being able to insert pictures might allow the writer to sort out the structure of the novel more.

Pictures in text are not planned, as mentioned in #815. This is considered out of scope for this app. The decision to not make it a general research/note-taking tool has to do with how much more complicated those are to implement.

Adding images as attachments will allow adding maps and diagrams to the notes, which is still useful. I think allowing to open images in the viewer on the right side is also easy to do. Other attachments would be opened externally.

**Comment by rosyhickey** [comment]
> > I saw the content of #815, and maybe it's a good idea as a plain text editor, except that being able to insert pictures might allow the writer to sort out the structure of the novel more.
> 
> Pictures in text are not planned, as mentioned in #815. This is considered out of scope for this app. The decision to not make it a general research/note-taking tool has to do with how much more complicated those are to implement.
> 
> Adding images as attachments will allow adding maps and diagrams to the notes, which is still useful. I think allowing to open images in the viewer on the right side is also easy to do. Other attachments would be opened externally.

Yes, if you open the image as an attachment on the right, it is also a better option. In addition, I would like to ask whether NW has an online synchronization function? Because if I write in two places, I need to send the entire file to my other computer so that I can continue writing on the other computer. It feels a bit troublesome to transfer the entire file package every time and then download it again to open it.

**Comment by vkbo** [comment]
Not a built-in one, no. That's also mentioned in #815. There are far too many security concerns and considerations with such a feature. However, the file structure of a novelWriter project is very easy for any standard file sync tool to handle. It should work on pretty much all of them as long as they don't have a weird pseudo-filesystem.

I sync my projects between desktop and laptop, and novelWriter was designed to be file sync friendly from day one.

**Comment by rosyhickey** [comment]
What you said makes sense.I‘m Looking forward to NW's attachment function coming online.
